modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/servlet/taglib/OpenGraphTopHeadDynamicInclude.java

```
@@ -0,0 +1,148 @@
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alicia García
+ */
+@Component(service = DynamicInclude.class)
+public class OpenGraphTopHeadDynamicInclude extends BaseDynamicInclude {
+
+	@Override
+	public void include(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		PrintWriter printWriter = httpServletResponse.getWriter();
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		StringBundler sb = new StringBundler(15);
+		Layout layout = themeDisplay.getLayout();
+
+		String languageId = themeDisplay.getLanguageId();
+
+		_appendOpenGraphTag(sb, "og:title", layout.getHTMLTitle(languageId));
+		_appendOpenGraphTag(
+			sb, "og:description", layout.getDescription(languageId));
+
+		_appendOpenGraphTag(sb, "og:locale", languageId);
+
+		try {
+			String canonicalURL = _portal.getCanonicalURL(
+				_portal.getCurrentCompleteURL(httpServletRequest), themeDisplay,
+				layout, false, false);
+
+			if (Validator.isNotNull(canonicalURL)) {
+				_appendOpenGraphTag(sb, "og:url", canonicalURL);
+
+				Map<Locale, String> alternateURLs = Collections.emptyMap();
+
+				Set<Locale> availableLocales = LanguageUtil.getAvailableLocales(
+					themeDisplay.getSiteGroupId());
+
+				if (availableLocales.size() > 1) {
+					alternateURLs = _portal.getAlternateURLs(
+						canonicalURL, themeDisplay, layout);
+				}
+
+				for (LayoutSEOLink layoutSEOLink :
+						LayoutSEOLinkManagerUtil.getLocalizedLayoutSEOLinks(
+							layout, _portal.getLocale(httpServletRequest),
+							canonicalURL, alternateURLs)) {
+
+					_appendOpenGraphTag(
+						sb, "og:locale:alternate", layoutSEOLink.getHrefLang());
+				}
+			}
+		}
+		catch (PortalException pe) {
```

Do not swallow exceptions. If we are not sure what to do when an error occurs, it is better to bubble it up (plus, it may help us to find bugs).